File: apps/backend/app/schemas/routing.py
from uuid import UUID

from pydantic import BaseModel

from app.models.enums import RequestUrgency


class RoutingHospitalPoint(BaseModel):
    hospital_id: UUID
    hospital_name: str
    city: str
    latitude: float
    longitude: float


class RoutingCandidate(BaseModel):
    source_hospital_id: UUID
    source_hospital_name: str
    source_city: str
    latitude: float
    longitude: float
    distance_km: float
    available_units: int
    available_volume_ml: int


class RoutingRecommendationResponse(BaseModel):
    request_id: UUID
    request_number: str
    urgency: RequestUrgency
    units_requested: int
    requesting_hospital: RoutingHospitalPoint
    candidates: list[RoutingCandidate]

File: apps/frontend/src/types/routing.ts
import type { RequestUrgency } from "@/types/domain";

export type { RequestUrgency } from "@/types/domain";

export interface RoutingHospitalPoint {
  hospital_id: string;
  hospital_name: string;
  city: string;
  latitude: number;
  longitude: number;
}

export interface RoutingCandidate {
  source_hospital_id: string;
  source_hospital_name: string;
  source_city: string;
  latitude: number;
  longitude: number;
  distance_km: number;
  available_units: number;
  available_volume_ml: number;
}

export interface RoutingRecommendationResponse {
  request_id: string;
  request_number: string;
  urgency: RequestUrgency;
  units_requested: number;
  requesting_hospital: RoutingHospitalPoint;
  candidates: RoutingCandidate[];
}
